lib/carto/named_maps/template.rb

```diff
@@ -183,7 +185,11 @@ def attributes_and_interactivity(layer_infowindow, layer_tooltip)
           }
         end
 
-        [attributes, interactivity.join(',')]
+        if layer_options[:interactivity]
+          interactivity = interactivity & layer_options[:interactivity].split(',')
+        end
+
+        [attributes, interactivity.uniq.join(',')]
```

Comment:
Why do you need uniq? Doesn't the and mask take care of that for you?

Reply:
`uniq` is needed in the case that `layer_options[:interactivity]` is not set.  Currently, `click_fields` and `hover_fields` could in theory both contain `cartodb_id`.

Reply:
Ah, I see! I thought that you were trying to remove duplicates from the '&'ing, which I believe is already a part of the `&`.

Maybe you could `uniq` right where you suspect the duplicates could be added? It adds to the "self documented" to assume that it's applied as needed.